dev-libs/rocm-comgr/files/rocm-comgr-6.1.0-llvm-18-compat.patch

```diff
@@ -0,0 +1,79 @@
+ROCm 6.0.0 and 6.0.2 releases use mix between LLVM 17 and 18
+forked as https://github.com/RadeonOpenCompute/llvm-project
+which makes some libraries compatible with LLVM 17,
+while other require LLVM 18.
+
+Backports:
+* https://github.com/ROCm/llvm-project/commit/6cbc4dc91dfeb1cf2295cb350866e0b3a07dfee4
+* https://github.com/ROCm/llvm-project/commit/179ec2e67bf882c6bccb27f81db3d80f7eb9946e
+* https://github.com/ROCm/llvm-project/commit/ee123c3d1706bc4346511b1a9032020782576350
+--- a/src/comgr-compiler.cpp
++++ b/src/comgr-compiler.cpp
+@@ -205,7 +205,11 @@ bool AssemblerInvocation::createFromArgs(AssemblerInvocation &Opts,
+   // Parse the arguments.
+   const OptTable &OptTbl = getDriverOptTable();
+ 
++#if LLVM_VERSION_MAJOR == 17
+   const unsigned IncludedFlagsBitmask = options::CC1AsOption;
++#else
++  llvm::opt::Visibility IncludedFlagsBitmask(options::CC1AsOption);
++#endif
+   unsigned MissingArgIndex, MissingArgCount;
+   InputArgList Args = OptTbl.ParseArgs(Argv, MissingArgIndex, MissingArgCount,
+                                        IncludedFlagsBitmask);
+@@ -1041,11 +1045,15 @@ amd_comgr_status_t AMDGPUCompiler::addCompilationFlags() {
+     Args.push_back("cl");
+     Args.push_back("-std=cl1.2");
+     Args.push_back("-cl-no-stdinc");
++    Args.push_back("-mllvm");
++    Args.push_back("-amdgpu-internalize-symbols");
```

Comment:
Sorry, I don't find where these two lines are from. Which problem does it solve? I can confirm that the rest of this patch are coming from upstream commits for clang-18.

Reply:
This fixes one test which fails with LLVM 18 (but succeeds with LLVM 17).
When it fails at line https://github.com/ROCm/llvm-project/blob/rocm-6.1.1/amd/comgr/test/name_expression_map_test.c#L127 comgr produces 3 symbols instead of 2. The extra symbol is `weak __oclc_ABI_version`, not internalized from opencl runtime. This is not present in upstream.

Reply:
Upstream has https://github.com/ROCm/llvm-project/commit/52161580a9c343813a5f46567864f8a98b655444 and https://github.com/ROCm/llvm-project/commit/27b8c1c1d04276e6911eadb97b4fbb8b23662e81 to fix this test, and I backported them in https://github.com/gentoo/gentoo/pull/36509/commits/983d0488bf153296d4a0543cc71f846f7e6f54ff#diff-4bc917a8689ee684b408248af6ba8ccd9dcee77d3e0cb429ecfc96c714e9e9ae

WDYT?

Reply:
If you backport change of test, I guess the correct code should look like
```
#if LLVM_VERSION_MAJOR == 17
// expect 2 symbols
#else
// expect 3 symbols
#endif
```
because `clang-17` still produces 2 symbols (you can check with `llvm-disasm`)